lib/osf-components/addon/components/moderators/add-modal/component.ts

```diff
@@ -0,0 +1,143 @@
+import UserModel from 'ember-osf-web/models/user';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+import Toast from 'ember-toastr/services/toast';
+import { ModeratorManager } from '../manager/component';
+import template from './template';
+
+interface InviteFormFields {
+    email: string;
+    fullName: string;
+    permissionGroup: PermissionGroup;
+}
+
+interface UserFormFields {
+    user: UserModel;
+    permissionGroup: PermissionGroup;
+}
+
+const InviteFormValidations: ValidationObject<InviteFormFields> = {
+    email: [
+        validatePresence({
+            presence: true,
+            type: 'empty',
+        }),
+        validateFormat({
+            type: 'email',
+            translationArgs: { description: 'This field' },
+        }),
+    ],
+    fullName: validatePresence({
+        presence: true,
+        type: 'empty',
+    }),
+    permissionGroup: validatePresence({
+        presence: true,
+        type: 'empty',
+    }),
+};
+
+const UserFormValidations: ValidationObject<UserFormFields> = {
+    user: validatePresence({
+        presence: true,
+        type: 'empty',
+    }),
+    permissionGroup: validatePresence({
+        presence: true,
+        type: 'empty',
+    }),
+};
+
+@tagName('')
+@layout(template)
+export default class AddModalComponent extends Component {
+    @service currentUser!: CurrentUserService;
+    @service toast!: Toast;
+    @service store!: DS.Store;
+
+    manager!: ModeratorManager;
+    @tracked shouldOpenAddDialog = false;
+
+    permissionOptions = [
+        PermissionGroup.Moderator,
+        PermissionGroup.Admin,
+    ];
+
+    inviteChangeset = buildChangeset({
+        fullName: null,
+        email: null,
+        permissionGroup: null,
+    }, InviteFormValidations);
+
+    userChangeset = buildChangeset({
+        user: null,
+        permissionGroup: null,
+    }, UserFormValidations);
+
+    @task({ withTestWaiter: true, restartable: true })
+    searchUser =
+    task(function *(this: AddModalComponent, name: string) {
+        try {
+            yield timeout(500);
+            return yield this.store.query('user', {
+                'filter[fullName]': name,
+            });
+        } catch (e) {
+            captureException(e);
+            this.toast.error(getApiErrorMessage(e));
+            return null;
+        }
+    });
+
+    @action
+    addUser() {
+        this.userChangeset.validate();
+        this.inviteChangeset.validate();
+        if (this.userChangeset.get('isValid')) {
+            this.manager.addUserAsModerator(
+                this.userChangeset.get('user'),
+                this.userChangeset.get('permissionGroup'),
+            );
+            this.closeAddModeratorDialog();
+        }
+        if (this.inviteChangeset.get('isValid')) {
```

Comment:
Doesn't a changeset with no changes evaluate as valid `isValid`? If that's correct, we should probably return early or only validate the changeset that has changes.